server/aesdsocket.c

```diff
@@ -60,10 +77,24 @@ int main(int argc, char *argv[])
         inet_ntop(client_addr.ss_family, get_address((struct sockaddr *)&client_addr), address_buffer, sizeof(address_buffer));
         syslog(LOG_USER, "Accepted connection from %s\n", address_buffer);
 
-        handle_client_connection(client_fd);
-        syslog(LOG_USER, "Closed connection from %s\n", address_buffer);
+        // Create a thread for the client connection
+        int *client_fd_ptr = malloc(sizeof(int));
+        *client_fd_ptr = client_fd;
+
+        pthread_t client_thread;
+        pthread_create(&client_thread, NULL, handle_client_connection, client_fd_ptr);
+
+        // Add the new thread to the linked list
+        struct thread_node *new_node = malloc(sizeof(struct thread_node));
+        new_node->thread_id = client_thread;
+        new_node->next = head;
+        head = new_node;
     }
 
+    // Clean up
+    cleanup_threads();
```

Comment:
@Sriram-Y what does it mean to only clean up threads here?  What happens if this software runs for a long time before exiting?